[Code Quality] Issue in lib/executor/instantiate/data.cpp

```
/home/runner/work/WasmEdge/WasmEdge/lib/executor/instantiate/data.cpp:37: error: Null Dereference
  pointer `MemInst` last assigned on line 35 could be null and is dereferenced at line 37, column 14.
  35.         auto *MemInst = getMemInstByIdx(StoreMgr, DataSeg.getIdx());
  36.         /// Check data fits.
  37.         if (!MemInst->checkAccessBound(
                   ^
  38.                 Offset, static_cast<uint32_t>(DataSeg.getData().size()))) {
  39.           spdlog::error(ErrCode::DataSegDoesNotFit);
```

--- a/lib/executor/instantiate/data.cpp
+++ b/lib/executor/instantiate/data.cpp
@@ -34,6 +34,7 @@ Expect<void> Executor::instantiate(Runtime::StoreManager &StoreMgr,
         /// Memory index should be 0. Checked in validation phase.
         auto *MemInst = getMemInstByIdx(StoreMgr, DataSeg.getIdx());
         /// Check data fits.
+        assert(MemInst != nullptr);
         if (!MemInst->checkAccessBound(
                 Offset, static_cast<uint32_t>(DataSeg.getData().size()))) {
           spdlog::error(ErrCode::DataSegDoesNotFit);
@@ -69,6 +70,7 @@ Expect<void> Executor::initMemory(Runtime::StoreManager &StoreMgr,
     if (DataSeg.getMode() == AST::DataSegment::DataMode::Active) {
       /// Memory index should be 0. Checked in validation phase.
       auto *MemInst = getMemInstByIdx(StoreMgr, DataSeg.getIdx());
+      assert(DataInst != nullptr);
       const uint32_t Off = DataInst->getOffset();
 
       /// Replace mem[Off : Off + n] with data[0 : n].

